subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/EclipseModelBuilder.java

```diff
@@ -262,6 +292,99 @@ private void populate(Project project) {
         }
     }
 
+    private Map<String, String> gatherClosedProjectPaths(Set<String> closedProjectNames) {
+        Map<String, String> closedProjectPaths = new HashMap<>();
+        for (ProjectState state : projectRegistry.getAllProjects()) {
+            ProjectInternal project = state.getOwner().getLoadedSettings().getGradle().getRootProject().project(state.getComponentIdentifier().getProjectPath());
```

Comment:
❌ Use `IdeArtifactRegistry` instead of reaching into other project's state. Add more information to it if necessary.